record/record.controller.js

```diff
@@ -450,6 +456,150 @@
+                    response.clickOkToDismiss = true;
+                    var reloadModalRows = function () {
+                        if (response.failedTupleData.length > 0) {
+                            // iterate over the set of successful ids and find them in selected rows, then remove them
+                            response.successTupleData.forEach(function (data) {
+                                // data is an object of key/value pairs for each piece of key information
+                                // { keycol1: val, keycol2: val2, ... }
+                                // TODO: what to do with multiple keys?
+                                // Object.keys(data).forEach(function (key) {
+                                var key = Object.keys(data)[0];
+                                var idx = searchPopupTableModel.selectedRows.findIndex(function (tuple) {
+                                    return tuple.data[key] == data[key];
+                                });
+
+                                searchPopupTableModel.selectedRows.splice(idx, 1);
+                                // });
+                            });
+                        } else {
+                            // if everything is successful, empty selected rows
+                            searchPopupTableModel.selectedRows = [];
+                        }
+
+                        recordTableUtils.update(searchPopupTableModel, true, true, true, false, logService.reloadCauses.BATCH_UNLINK);
+                    }
+
+                    // TODO: - improve partial success and use TRS to check delete rights before giving a checkbox
+                    //       - some errors could have been because of row level security
+                    ErrorService.handleException(response, false, false, reloadModalRows, reloadModalRows);
+                }
+
+                // error response function for deleteBatchAssociationTuples
+                var deleteError = function (err) {
+                    $rootScope.showSpinner = false;
+                    // errors that land here would be execution of code errors
+                    // if a deletion fails/errors, that delete request is caught by ermrestJS and returned
+                    //   as part of the deleteErrors object in the success cb
+                    // NOTE: if one of the identifying values is empty or null, an error is thrown here
+                    $log.warn(err);
+                    throw err;
+                }
+
+                if (!CONFIRM_DELETE) {
+                    return leafReference.deleteBatchAssociationTuples(tableModel.parentTuple, tuples).then(deleteResponse).catch(deleteError);
+                }
+
+                // log the opening of the confirm delete modal
+                logService.logClientAction({
+                    action: logService.getActionString(logService.logActions.UNLINK_INTEND),
+                    stack: logService.getStackObject()
```

Comment:
You should use the `logStack` that you created above. `logService.getStackObject()` will return the stack object attached to the `$rootScope` which doesn't include the related entity data.